DesignDoc.md

```diff
@@ -0,0 +1,157 @@
+
+## 2. Top Questions to Resolve in Review
+
+
+1. Working with multiple gsi.
+2. Figuring how to make a search function. (looking up certain movies by title, director or actor)
+3. Creating endpoint for movie API.
+4. Should I incorporate a follow feature for other movie buffs
+5. Find a way to link socials or sign in with socials 
+
+
+## 3. Use Cases
+
+_This is where we work backwards from the customer and define what our customers would like to do (and why). You may also include use cases for yourselves (as developers), or for the organization providing the product to customers._
+
+1. As a user, I want to be able to create an account on Reel Deal so that I can save my favorite movies and review them.
+
+2. As a user, I want to be able to search for movies by title, director, or actor so that I can easily find the films I'm looking for.
+
+3. As a user, I want to be able to rate and review movies on Reel Deal so that I can share my thoughts with others.
+
+4. As a user, I want to be able to follow other users on Reel Deal so that I can see their reviews and ratings, and discover new movies.
+
+5. As a user, I want to be able to save movies to my "watchlist" on Reel Deal so that I can keep track of the films I want to watch later.
+
+6. As a user, I want to be able to create custom lists on Reel Deal so that I can organize my favorite films by genre, director, or any other criteria.
+
+7. As a user , I want to be able to view statistics and data on movies on Reel Deal so that I can learn more about the films I watch.
+
+8. As a user, I want to be able to view trailers and posters for movies on Reel Deal so that I can get a better idea of what a film is about before watching it.
+
+
+### 4.1. In Scope
+
+* looking at reviews from others people
+* search movies by title director and actor
+* show popular movies to leave reviews on.
+* Adding, updating, deleting, and viewing reviews.
+
+
+### 4.2. Out of Scope
+
+* following other users .
+
+
+# 5. Proposed Architecture Overview
+
+_Describe broadly how you are proposing to solve for the requirements you described in Section 2. This may include class diagram(s) showing what components you are planning to build. You should argue why this architecture (organization of components) is reasonable. That is, why it represents a good data flow and a good separation of concerns. Where applicable, argue why this architecture satisfies the stated requirements._
+
+We will use API Gateway and Lambda to create endpoints that will handle the creation, updating, and retrieval of movies and reviews. 
+
+We will store movie review information in a table in DynamoDB.
+
+ReelDeal will also provide a web interface for users. A main page providing a variety of options to add, update and reminisce of all the movies you watched and maybe some you warn others not to watch.
```

Comment:
A general comment ... there's a lot described here that implies a good amount of UI functionality. As you work through building this out make sure to focus on the backend/API functionality _first_, and defer doing a lot of UI/UX design work until later in the process.